WooCommerce/Classes/ViewRelated/Notifications/ReviewsCoordinator.swift

```diff
@@ -0,0 +1,110 @@
+
+/// Coordinator for the Reviews tab.
+///
+final class ReviewsCoordinator: Coordinator {
+    var navigationController: UINavigationController
+
+    private let pushNotificationsManager: PushNotesManager
+    private let storesManager: StoresManager
+    private let noticePresenter: NoticePresenter
+    private let switchStoreUseCase: SwitchStoreUseCaseProtocol
+
+    private var observationToken: ObservationToken?
+
+    private let willPresentReviewDetailsFromPushNotification: () -> Void
+
+    init(pushNotificationsManager: PushNotesManager = ServiceLocator.pushNotesManager,
+         storesManager: StoresManager = ServiceLocator.stores,
+         noticePresenter: NoticePresenter = ServiceLocator.noticePresenter,
+         switchStoreUseCase: SwitchStoreUseCaseProtocol,
+         willPresentReviewDetailsFromPushNotification: @escaping () -> Void) {
+
+        self.pushNotificationsManager = pushNotificationsManager
+        self.storesManager = storesManager
+        self.noticePresenter = noticePresenter
+        self.switchStoreUseCase = switchStoreUseCase
+        self.willPresentReviewDetailsFromPushNotification = willPresentReviewDetailsFromPushNotification
+
+        self.navigationController = WooNavigationController(rootViewController: ReviewsViewController())
+    }
+
+    convenience init(willPresentReviewDetailsFromPushNotification: @escaping () -> Void) {
+        let storesManager = ServiceLocator.stores
+        self.init(storesManager: storesManager,
+                  switchStoreUseCase: SwitchStoreUseCase(stores: storesManager),
+                  willPresentReviewDetailsFromPushNotification: willPresentReviewDetailsFromPushNotification)
+    }
+
+    deinit {
+        observationToken?.cancel()
+    }
+
+    func start() {
+        observationToken = pushNotificationsManager.inactiveNotifications.subscribe { [weak self] in
+            self?.handleInactiveNotification($0)
+        }
+    }
+
+    private func handleInactiveNotification(_ notification: PushNotification) {
+        guard notification.kind == .comment else {
+            return
+        }
+
+        let action = ProductReviewAction.retrieveProductReviewFromNote(noteID: Int64(notification.noteID)) { [weak self] result in
+            guard let self = self else {
+                return
+            }
+
+            switch result {
+            case .failure:
+                self.noticePresenter.enqueue(notice: Notice(title: Localization.failedToRetrieveNotificationDetails))
+            case .success(let parcel):
+                guard let siteID = parcel.note.meta.identifier(forKey: .site) else {
+                    self.noticePresenter.enqueue(notice: Notice(title: Localization.failedToRetrieveNotificationDetails))
+                    return
+                }
+
+                // Switch to the correct store first if needed
+                self.switchStoreUseCase.switchStore(with: Int64(siteID)) { [weak self] siteChanged in
```

Comment:
Good eye. It is a new behavior. The `UseCase` was added during  https://github.com/woocommerce/woocommerce-ios/pull/2013.

>  just thought that we might not have a way to test this flow yet

We can test this by modifying the code. I guess I didn't think it was too important to this since, in #2013, we tried to make sure that the `UseCase` will do the right thing for us. But yeah, you're right, it's not a bad idea to this too.

I copied @pmusolino's testing steps in #2013 and updated the description. 😁